fastly/ngwaf/v1/workspaces/api_response.go

```diff
@@ -0,0 +1,47 @@
+package workspaces
+
+import "time"
+
+// Workspace is the API response structure for the create, update, and get operations.
+type Workspace struct {
+	// WorkspaceID is the workspace identifier (UUID).
+	WorkspaceID string `json:"id"`
+	// Name is the name of the workspace.
+	Name string `json:"name"`
+	// Description is the description of the workspace.
+	Description string `json:"description"`
+	// Mode is the mode of the workspace.
+	Mode string `json:"mode"`
+	// AttackSignalThresholds are the parameters for system site alerts.
+	AttackSignalThresholds AttackSignalThresholds `json:"attack_signal_thresholds"`
+	// IPAnonymization is the selected option to anonymize IP addresses.
+	IPAnonymization string `json:"ip_anonymization"`
+	// CreatedAt is the date and time in ISO 8601 format.
+	CreatedAt time.Time `json:"created_at"`
+	// UpdatedAt is the date and time in ISO 8601 format.
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+// AttackSignalThresholds are the parameters for system site alerts.
+type AttackSignalThresholds struct {
+	OneMinute  int  `json:"one_minute"`
+	TenMinutes int  `json:"ten_minutes"`
+	OneHour    int  `json:"one_hour"`
+	Immediate  bool `json:"immediate"`
+}
```

Comment:
Let me know if we should have one combined struct for the `Create()`, `Update()` inputs as well as the response struct.

Reply:
I don't think any of them can be combined directly, but it might be possible to create a struct which has a common subset of all of them and use embedding to include it into those other structures. I don't have a strong opinion either way.